Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -15,41 +15,112 @@
 // permissions and limitations under the License.
 
 using System;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using Microsoft.PythonTools.Debugger.DebugEngine;
+using Microsoft.PythonTools.Infrastructure;
 using Microsoft.VisualStudio.Debugger.Interop.VSCodeDebuggerHost;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.PythonTools.Debugger {
     public sealed class DebugAdapterProcess : ITargetHostProcess {
-        private readonly Process _process;
-        private DebugAdapterProcess(ProcessStartInfo psi) {
+        private const int _debuggerConnectionTimeout = 5000; // 5000 ms
+        private const int _connectionCloseTimeout = 5000; // 5000 ms
+
+        private Process _process;
+        private readonly Guid _processGuid;
+        private PythonDebugOptions _debugOptions;
+        private string _interpreterOptions;
+        private int _listenerPort = -1;
+        private Stream _stream;
```

Comment:
How this stream get closed?

Reply:
On Process exit, we close the socket and the stream.